content/lua/concepts/strings/terms/find/find.md

```diff
@@ -0,0 +1,176 @@
+
+The results of the above functions are:
+
+```shell
+1
+5
+1  5
+7
+11
+7  11
+```
+
+- `hello` starts at index 1, and ends at index 5 in the `sentence` string. Due to multiple assignments, `a` saves the value `1`, and `b` saves the second value `5`.
+- `world` starts at index 7, and ends at index 11 in the `sentence` string. Due to multiple assignments, `y` saves the first value `7`, and `z` saves the second value `11`.
+
+## Example 3
+
+Use `.find()` with `init` (Initial position) parameter:
+
+```lua
+sentence1 = 'hello world'
+search = string.find(sentence1, "l", 5)
+print(search)  -- Output: 10
+
+sentence2 = 'world world'
+a, b = string.find(sentence2, "orl", 5)
+print(a)  -- Output: 8
+print(b)  -- Output: 10
+
+print(a, b)  -- Output: 8  10
+```
+
+The above example uses the initial position parameter to indicate where the search shout start.
+
+The results of the above functions are:
+
+```shell
+10
+8
+10
+8  10
+```
+
+- `l` has multiple instances within the `sentence` string. Due to the init parameter, the search starts at index 5, therefore the first instance of `l` from index 5 will be at starting index `10`.
+- `orl` has multiple instances within the `sentence` string. Due to the init parameter, the search starts at index 5, therefore the first instance of `orl` from index 5 will be at starting index `8`, and ends at index `10`.
+
+## Example 4
+
+Use `.find()` with `pattern` (pattern matching) parameter:
+
+```lua
+-- With pattern matching
+sentence = "hello world"
+a, b = string.find(sentence, "l+", 1, false)
+print(a)
+print(b)
+
+-- Without pattern matching
+y, x = string.find("hello world", "l+", 1, true)
+print(y)
+print(z)
+```
+
+The above example uses the pattern matching parameter to determine if pattern matching characters can be used as search terms.
+
+The results of the above functions are:
+
+```shell
+3
+4
+nil
+nil
+```
+
+- In the first example, `l+` uses '+' to match one or more repetitions of the previous character 'l'. Due to the `false` boolean value, pattern matching is being used, which results the outputs of `a` and `b` resulting to 3 and 4 respectively.
+- In the second example, `l+` is being treated as a literal string, which doesn't exist in the string `hello world`. Due to the `true` boolean value, pattern matching is disabled, which results the outputs of `nil` in both `y` and `z`, as the string `l+` does not exist in the string `hello world`.
+
+[^1]: The special pattern matching characters allow for lua to use specified characters in `searchString` to make a pattern matching search. This allows for complex search patterns, which can enhance the flexibility of your searches. Pattern matching is similar to the concept [Regular Expressions](https://www.codecademy.com/resources/docs/general/regular-expressions), although Lua's pattern matching is not the same as regular expression, as it is more limited, and uses a different syntax.
```

Comment:
```suggestion
> **Note:** [^1]: The special pattern matching characters allow for lua to use specified characters in `searchString` to make a pattern matching search. This allows for complex search patterns, which can enhance the flexibility of your searches. Pattern matching is similar to the concept [Regular Expressions](https://www.codecademy.com/resources/docs/general/regular-expressions), although Lua's pattern matching is not the same as regular expression, as it is more limited, and uses a different syntax.
```

Reply:
if we are converting this into a note, doesn't it make more sense to have it between the explanations between the syntax explanation, and the first example?